tests/rptest/tests/datalake/catalog_service_factory.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2025 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.catalog_service import CatalogType, CatalogService
+from rptest.services.nessie_catalog import NessieCatalog
+from rptest.services.apache_iceberg_catalog import IcebergRESTCatalog
+
+from typing import List
+
+# TODO: Nessie is not supported here yet. Add it back once
+# required datalake authentication changes have been made.
+SUPPORTED_CATALOG_TYPES = [CatalogType.REST_JDBC, CatalogType.REST_HADOOP]
+
+
+def filesystem_catalog_type() -> CatalogType:
+    return CatalogType.REST_HADOOP
+
+
+def supported_catalog_types() -> List[CatalogType]:
+    return SUPPORTED_CATALOG_TYPES
+
+
+def make_catalog_service_for_type(catalog_type: CatalogType, test_ctx,
+                                  cloud_storage_bucket: str,
+                                  catalog_name: str) -> CatalogService:
```

Comment:
nit: update to `warehouse_name` to follow suit with the rest of the name changes?